### PR TITLE
Use organization for PipelineCreateRequest

### PR DESCRIPTION
**Pull request by alvarolobato**
# Description
Follow up - https://github.com/jenkinsci/blueocean-plugin/pull/1278#discussion_r131783811

Tracked as JENKINS-46125

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java [hunk 1]
```diff
@@ -75,11 +75,11 @@ public GithubPipelineCreateRequest(String name, BlueScmConfig scmConfig) {
     @SuppressWarnings("unchecked")
     @SuppressFBWarnings(value = "REC_CATCH_EXCEPTION", justification = "Runtime exception is thrown from the catch block")
     @Override
-    public BluePipeline create(Reachable parent) throws IOException {
+    public BluePipeline create(Reachable parent, @Nonnull BlueOrganization organization) throws IOException {
         Preconditions.checkNotNull(parent, "Parent passed is null");
         Preconditions.checkNotNull(getName(), "Name provided was null");
 
-        User authenticatedUser = checkUserIsAuthenticatedAndHasItemCreatePermission();
+        User authenticatedUser = checkUserIsAuthenticatedAndHasItemCreatePermission(organization);
```
Comment (alvarolobato) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java]: I was in doubt if it was best passing the organization to all these methods or have it in a field. But given that we can't initialize it in the constructor, it seems to be safer this way.

File: blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineCreateRequest.java [hunk 2]
```diff
@@ -24,19 +23,11 @@ public void setName(String name) {
...
-        this.organization = organization;
-    }
-
     /**
      * Create an instance of {@link BluePipeline} from the given request.
      *
      * @return created pipeline
      */
-    public abstract @CheckForNull BluePipeline create(Reachable parent) throws IOException;
+    public abstract @CheckForNull BluePipeline create(Reachable parent, BlueOrganization organization) throws IOException;
```
Comment (vivek) [inline blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineCreateRequest.java]: Most places we have Reachable as last parameter and also lets call out organization and parent parameter `@Nonnull`.

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java (hunk 1 above)
Comment (vivek) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java]: that sounds fine.

**Comment by michaelneale** [comment]
once build is green this is :bee: from me - but don't merge until we confirm editor live test is ok @vivek

**Comment by alvarolobato** [comment]
@vivek comments addressed

**Comment by vivek** [comment]
@alvarolobato there appears to be conflict.

**Comment by alvarolobato** [comment]
Rebase done. Lets wait for the build

**Comment by michaelneale** [comment]
via some miracle the build is green (been having flaky live tests at certain times of day lately..) so up to @vivek to merge this tomorrow if it suits.

**Comment by alvarolobato** [comment]
👏

**Pull request merged**